test/chemistry/test_adapt_vqe.py

```diff
@@ -0,0 +1,141 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Test of the Adaptive VQE ground state calculations """
+import unittest
+from test.chemistry import QiskitChemistryTestCase
+
+from qiskit.chemistry.drivers import PySCFDriver, UnitsType
+from qiskit.providers.basicaer import BasicAer
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.algorithms import VQE
+from qiskit.chemistry import QiskitChemistryError
+from qiskit.chemistry.components.variational_forms import UCCSD
+from qiskit.chemistry.components.initial_states import HartreeFock
+from qiskit.aqua.components.optimizers import L_BFGS_B
+from qiskit.chemistry.ground_state_calculation import AdaptVQE, VQEUCCSDFactory
+from qiskit.chemistry.qubit_transformations import FermionicTransformation
+
+
+class TestAdaptVQE(QiskitChemistryTestCase):
+    """ Test Adaptive VQE Ground State Calculation """
+
+    def setUp(self):
+        super().setUp()
+
+        try:
+            self.driver = PySCFDriver(atom='H .0 .0 .0; H .0 .0 0.735',
+                                      unit=UnitsType.ANGSTROM,
+                                      basis='sto3g')
+        except QiskitChemistryError:
+            self.skipTest('PYSCF driver does not appear to be installed')
+            return
+
+        self.expected = -1.85727503
+
+        self.transformation = FermionicTransformation()
+
+    def test_default(self):
+        """ Default execution """
+        solver = VQEUCCSDFactory(QuantumInstance(BasicAer.get_backend('statevector_simulator')))
+        calc = AdaptVQE(self.transformation, solver)
+        res = calc.compute_groundstate(self.driver)
+        self.assertAlmostEqual(res.electronic_energy, self.expected, places=6)
+
+    def test_custom_minimum_eigensolver(self):
+        """ Test custom MES """
+        # Note: the VQEUCCSDFactory actually allows to specify an optimizer through its constructor.
+        # Thus, this example is quite far fetched but for a proof-of-principle test it still works.
+        class CustomFactory(VQEUCCSDFactory):
```

Comment:
Since the 'standard' VQEUCCSDFactory has the capability to be configured perhaps it is better to more directly use that here now, since it can get more testing, rather than a custom extension.

Reply:
That's exactly why I left the comment above. I agree that the `VQEUCCSDFactory` needs unittests itself.
But that should not be part of this file.

Reply:
Sure, it could also directly extend MESFactory if you want to test with a CustomFactory that things work as expected. We ought to have a CustomFactory that does not have VQE and/or UCCSD in it to check that it fails as expected when this is not the case